models/foreignreference.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2022 Gitea. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"code.gitea.io/gitea/models/db"
+)
+
+// ForeignReference represents external references
+type ForeignReference struct {
+	LocalID   int64  `xorm:"INDEX"`
```

Comment:
If the `LocalID` is the `issue index`, it's better to be called `LocalIssueIndex`, we should avoid introducing ambiguous concepts as much as possible.

Reply:
And about the database index, how should the `ForeignReference` be queried? I guess we should query it with `RepoID=? AND LocalIssueIndex=?`, so I think we only need the two indices:

* INDEX (RepoID, LocalIssueIndex)
* UNIQUE (RepoID, ForeignID, Type)